lib/suppose-stream.js

```diff
@@ -55,7 +56,7 @@ function SupposeStream(options) {
     switch(typeof condition)
     {
       case 'string':
-        match = buffer.endsWith(condition)
+        match = new RegExp('[^\\S\\r\\n]*' + escape(condition) + '[^\\S\\r\\n]*').test(buffer);
```

Comment:
I was asking to fully remove the initial `'[^\\S\\r\\n]*' + ` block, since it can be ignored due to the `*` repetition, it's mostly useless and only make the regular expression more complex and CPU consuming.